sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringServiceImpl.java

```diff
@@ -235,6 +280,23 @@ public Mono<SpringService> createResourceAsync() {
         if (isInCreateMode()) {
             createOrUpdate = manager().serviceClient().getServices()
                 .createOrUpdateAsync(resourceGroupName(), name(), innerModel());
+            if (isEnterpriseTier()) {
+                createOrUpdate = createOrUpdate
+                    // initialize build service agent pool
+                    .flatMap(inner ->
+                        manager().serviceClient().getBuildServiceAgentPools().updatePutAsync(
+                            resourceGroupName(),
+                            name(),
+                            Constants.DEFAULT_TANZU_COMPONENT_NAME,
+                            Constants.DEFAULT_TANZU_COMPONENT_NAME,
+                            new BuildServiceAgentPoolResourceInner()
+                                .withProperties(
+                                    new BuildServiceAgentPoolProperties()
+                                        .withPoolSize(
+                                            new BuildServiceAgentPoolSizeProperties()
+                                                .withName("S1"))) // S1, S2, S3, S4, S5.
```

Comment:
So this `S1` etc. seems to be configurable in future PR?

And for enterprise, does this mean it now had to send 2 PUT (or more if gitConfig set)?

Reply:
It is configurable but I don't provide support for it. I can provide support in future PR.
> And for enterprise, does this mean it now had to send 2 PUT (or more if gitConfig set)?

Yes it does. Without this step, build service will not have a configured agent pool and future build will fail..
And for now, default `Configuration Service` will always be created regardless of whether git config is set as it's the default behavior of both portal and cli.